openerp/addons/base/migrations/7.0.1.3/pre-migration.py

```diff
@@ -176,11 +183,24 @@ def create_users_partner(cr):
     cr.execute(
         "SELECT id, name, active FROM res_users "
         "WHERE partner_id IS NULL")
+    warning_installed = is_module_installed(cr, 'warning')
     for row in cr.fetchall():
-        cr.execute(
-            "INSERT INTO res_partner "
-            "(name, active) "
-            "VALUES(%s,%s) RETURNING id", row[1:])
+        if warning_installed:
+            sql_insert_partner = (
+                "INSERT INTO res_partner "
+                "(name, active,picking_warn,sale_warn,"
+                "invoice_warn,purchase_warn) "
+                "VALUES(%s,%s, %s, %s, %s, %s) RETURNING id")
+            sql_args = (
+                row[0], row[1],
+                'no-message', 'no-message', 'no-message', 'no-message')
```

Comment:
You don't need to pass 'no-message' as parameters. You can hard-code them into SQL. This way, you can save the SQL before the loop in a variable, and always use the same independently if the module is installed or not.